packages/features/eventtypes/components/EditWeightsForAllTeamMembers.tsx

```diff
@@ -0,0 +1,304 @@
+                  }
+                  if (e.key === "Escape") {
+                    setIsEditing(false);
+                  }
+                }}
+              />
+              <span className="text-default border-default bg-muted flex h-7 items-center rounded-r-sm border border-l-0 px-2 text-sm">
+                %
+              </span>
+            </div>
+          </div>
+        ) : (
+          <button
+            className="text-emphasis hover:bg-subtle decoration-emphasis flex h-7 items-center rounded-sm px-2 text-sm underline underline-offset-4"
+            onClick={() => setIsEditing(true)}>
+            {member.weight ?? 100}%
+          </button>
+        )}
+      </div>
+    </div>
+  );
+};
+
+interface Props {
+  teamMembers: TeamMember[];
+  value: Host[];
+  onChange: (hosts: Host[]) => void;
+  assignmentState?: AssignmentState;
+}
+
+export const EditWeightsForAllTeamMembers = ({ teamMembers, value, onChange, assignmentState }: Props) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const { t } = useLocale();
+  const [searchQuery, setSearchQuery] = useState("");
+  const [localWeights, setLocalWeights] = useState<Record<string, number>>(() =>
+    teamMembers.reduce<Record<string, number>>((acc, member) => {
+      acc[member.value] = member.weight ?? 100;
+      return acc;
+    }, {})
+  );
+  const [uploadErrors, setUploadErrors] = useState<Array<{ email: string; error: string }>>([]);
+  const [isErrorsExpanded, setIsErrorsExpanded] = useState(true);
+
+  const handleWeightChange = (memberId: string, weight: number) => {
+    setLocalWeights((prev) => ({ ...prev, [memberId]: weight }));
+  };
+
+  const handleSave = () => {
+    const updatedValue = value.map((host) => ({
+      ...host,
+      weight: localWeights[host.userId.toString()] ?? host.weight ?? 100,
+    }));
+    onChange(updatedValue);
+    setIsOpen(false);
+    showToast(t("weights_updated_successfully"), "success");
+  };
+
+  const handleDownloadCsv = () => {
+    const csvData = teamMembers.map((member) => ({
+      id: member.value,
+      name: member.label,
+      email: member.email,
+      weight: member.weight ?? 100,
+    }));
+    downloadAsCsv(csvData, "team-members-weights.csv");
+  };
+
+  const handleUploadCsv = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (!e.target.files?.length) return;
+
+    const file = e.target.files[0];
+    const reader = new FileReader();
+
+    reader.onload = async (event) => {
+      if (!event.target?.result) return;
+
+      const csvData = event.target.result as string;
+      const lines = csvData.split("\n");
+      const newWeights: Record<string, number> = { ...localWeights };
+      const newErrors: Array<{ email: string; error: string }> = [];
+
+      // Skip header row
+      for (let i = 1; i < lines.length; i++) {
+        const line = lines[i].trim();
+        if (!line) continue;
+
+        const [, , email, weightStr] = line.split(",");
```

Comment:
We expect the format of the downloaded CSV. In the future we can rely on checking headers for a ID/EMAIL + weight so any CSV/sheets can be used